x-pack/plugins/apm/server/lib/service_map/get_max_anomalies.ts

```diff
@@ -0,0 +1,188 @@
+  environment?: string;
+}) {
+  const { ml, start, end } = setup;
+
+  if (!ml) {
+    logger.warn('Anomaly detection plugin is not available.');
+    return DEFAULT_VALUE;
+  }
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (!mlCapabilities.mlFeatureEnabledInSpace) {
+    logger.warn('Anomaly detection feature is not enabled for the space.');
+    return DEFAULT_VALUE;
+  }
+  if (!mlCapabilities.isPlatinumOrTrialLicense) {
+    logger.warn(
+      'Unable to create anomaly detection jobs due to insufficient license.'
+    );
+    return DEFAULT_VALUE;
+  }
+
+  let mlJobIds: string[] = [];
+  try {
+    mlJobIds = await getMLJobIds(ml, environment);
+  } catch (error) {
+    logger.error(error);
+    return DEFAULT_VALUE;
+  }
+
+  const params = {
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                result_type: 'record',
+              },
+            },
+            {
+              terms: {
+                job_id: mlJobIds,
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: start,
+                  lte: end,
+                  format: 'epoch_millis',
+                },
+              },
+            },
+            {
+              terms: {
+                by_field_value: [TRANSACTION_REQUEST, TRANSACTION_PAGE_LOAD],
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        service_name: {
+          terms: {
+            field: 'partition_field_value',
+          },
+          aggs: {
+            top_score: {
+              top_hits: {
+                sort: {
+                  record_score: 'desc',
+                },
+                _source: {
+                  includes: ['actual', 'job_id', 'by_field_value'],
+                },
+                size: 1,
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+  const response = await ml.mlSystem.mlAnomalySearch(params);
+  return {
+    mlJobIds,
+    maxAnomalies: transformResponseToMaxAnomalies(
+      response as MaxAnomaliesAggResponse
+    ),
+  };
+}
+
+interface MaxAnomaliesAggResponse {
+  aggregations: {
+    service_name: {
+      buckets: Array<{
+        key: string;
+        top_score: {
+          hits: {
+            hits: Array<{
+              sort: [number];
+              _source: {
+                actual: [number];
+                job_id: string;
+                by_field_value: string;
+              };
+            }>;
+          };
+        };
+      }>;
+    };
+  };
+}
+
+function transformResponseToMaxAnomalies(
+  response: MaxAnomaliesAggResponse
+): Array<{
+  [SERVICE_NAME]: string;
+  maxAnomaly: MaxAnomaly;
+}> {
+  const services = response.aggregations.service_name.buckets.map(
+    ({ key: serviceName, top_score: topScoreAgg }) => {
+      return {
+        [SERVICE_NAME]: serviceName,
```

Comment:
I don't think the constants serve any purpose here
```suggestion
        serviceName,
```

Reply:
Btw. instead of returning an array it might be better to return an object `Record<string, MaxAnomaly>` where key is the service name